tally/src/buckets.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "tally/buckets.h"
+
+#include <stdexcept>
+
+#include "tally/buckets_iterator.h"
+#include "tally/src/buckets_calculator.h"
+
+namespace tally {
+
+Buckets::Buckets(Buckets::Kind kind, BucketsCalculator calculator, uint64_t num)
+    : kind_(kind), calculator_(calculator), num_(num) {
+  if (num == 0) {
+    throw std::invalid_argument("Number of buckets cannot be zero");
+  }
+}
+
+Buckets Buckets::LinearValues(double start, double width, uint64_t num) {
```

Comment:
Just curious, in some places you do `std::int64_t` and in others you do `uint64_t`. Sort of a nit, but would it make more sense to be consistent across the board?

Reply:
That's a good catch,  I'll fix.